vendor/github.com/vektah/gqlparser/v2/ast/decode.go

```diff
@@ -0,0 +1,198 @@
+package ast
+
+import (
+	"encoding/json"
+)
+
+func UnmarshalSelectionSet(b []byte) (SelectionSet, error) {
+	var tmp []json.RawMessage
+
+	if err := json.Unmarshal(b, &tmp); err != nil {
+		return nil, err
+	}
+
+	var result = make([]Selection, 0)
+	for _, item := range tmp {
+		var field Field
+		if err := json.Unmarshal(item, &field); err == nil {
```

Comment:
🤔 Under which conditions would this `err` be non-nil? I'm wondering because Golang would still return `nil` if none of the fields matched: https://go.dev/play/p/EmsX1UY5GZR

So I'm not sure I understand how we'd ever unmarshal FragmentSpread and InlineFragment here.

Reply:
I'm a bit worried about this, so I'll take a second look at it.